### ISSUE TITLE
[dev.icinga.com #8460] Ensure that id attributes are unique

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1394]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/8460

**Created by elippmann on 2015-02-17 09:07:42 +00:00**

Assignee: _(none)_
Status: _Resolved_  (closed on _2015-02-25 17:20:03 +00:00_)
Target Version: _2.0.0-beta3_
Last Update: _2015-02-25 17:20:03 +00:00 (in Redmine)_

---

http://www.w3.org/TR/2013/NOTE-WCAG20-TECHS-20130905/H93 wrote:
> The objective of this technique is to avoid key errors that are known to cause problems for assistive technologies when they are trying to parse content that has the same id attribute on different elements. These errors can be avoided by making sure the Web page does not have duplicate id values. This can be done manually or by using HTML's mechanism to specify the technology and technology version, and validating the document for this condition. There are several validators that the developer can use; validation reports generally mention this type of error. The document type declaration is not strictly necessary for this type of evaluation, but specifying the document type declaration makes it easier to use a validator.

**Changesets**

_2015-02-25 17:00:28 +00:00_ by _mjentsch_ 89451f30864c01ceedb2fae97355a4835486b4db

    Ensure that form ids are unique
    
    Add an unique prefix to each Form- or FormElement id, unless id protection is disabled explicitly, to prevent id collisions between different containers.
    
    fixes #8460

_2015-02-27 08:01:46 +00:00_ by _jmeyer_ 48c3199a2ddc46b27468150874d1bb800c91d5d4

    Form: Fix id protection
    
    Altering element ids upon rendering makes it impossible to reference
    them elsewhere. Zend_Form::getId() should be preferred over getName()
    as it falls automatically back to getName() in case there is no id attribute
    set. Setter are required to return their instance. Overwriting methods
    without calling the parent (getId) without any valid (or at least
    documented) reason is very bad practice.
    
    refs #8460
